aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -80,4 +80,20 @@ Tensor to(const Tensor& self, const Tensor& other, bool non_blocking, bool copy)
   return to_impl(self, options, non_blocking);
 }
 
+Tensor to_dense_backward(const Tensor& grad, const Tensor& input_) {
+  AT_ASSERT(input_.layout() != c10::kStrided);
+  if (input_.is_sparse()) {
+    auto input = input_.coalesce();
+    return grad.sparse_mask(at::SparseTensorRef(input));
+  } else {
+    AT_ASSERT(input_.layout() == c10::kMkldnn);
```

Comment:
nit: can you make this an else if with the layout, and then make the else an AT_ERROR with a nice error message?  Since we will be adding more layouts, presumably.

Reply:
@gchanan Fixed.